### ISSUE TITLE
why.... guys, just, WHY

### ISSUE THREAD

**Issue by piotrmasior** [run-llama/llama_index#9119]
![image](https://user-images.githubusercontent.com/1107041/285324461-0d103e73-29f3-49c4-841e-3b875dc3fa1f.png)

you produced here dead code

_Originally posted by @piotrmasior in https://github.com/run-llama/llama_index/pull/7343#discussion_r1403791939_

**Comment by logan-markewich** [comment]
What is the issue here?

**Comment by logan-markewich** [comment]
Just re-iterating from the thread:

```
is_formatted = kwargs.pop("formatted", False)
if not is_formatted:
    prompt = self.completion_to_prompt(prompt)
```

If you call chat() or stream_chat() then it will not enter this conditional

If you call complete() or stream_complete() (without formatted=True in kwargs) then it will enter the conditional.

No dead code? We dont want to call/stack messages_to_prompt() and completion_to_prompt() on top of each other, since they are different functions with different logic

**Comment by piotrmasior** [comment]
and if you want to call chat() or stream_chat() with formatted=False, you are screwed

**Comment by piotrmasior** [comment]
my arguments, as double copy in all places is new github UI approach...
https://github.com/run-llama/llama_index/pull/7343#discussion_r1403803255
https://github.com/run-llama/llama_index/pull/7343#discussion_r1403812602

**Comment by logan-markewich** [comment]
This isn't a bug, it's working as intended

**Comment by piotrmasior** [comment]
> This isn't a bug, it's working as intended

I rarely do this, but man, are you some kind of special? 
this exact change https://github.com/run-llama/llama_index/pull/7343/files/a2debb75d030ed33f82bc3286b837782a4d7724f#diff-9b76b57fc39aa4c4e4c6a3ae47ee15dc5f9d38a5e15354aebf20bfcd64902d35R184 is overshadowing the `formatted` variable, you no longer have possibility to change it when you do a method call. WHY are you trying to say it is not a problem when you are removing flexability ?

what is even more alarming at that line: https://github.com/run-llama/llama_index/pull/7343/files/a2debb75d030ed33f82bc3286b837782a4d7724f#diff-9b76b57fc39aa4c4e4c6a3ae47ee15dc5f9d38a5e15354aebf20bfcd64902d35R191

you are generally do not understand how kwargs work, let me show:

```
def old_good_function(prompt, **kwargs):
    print(f"My kwargs from {prompt}: {kwargs}")
    formatted = kwargs.pop("formatted", False)
    print("My kwargs can have formatted value:", formatted)

def your_stupid_definition(prompt, formatted=True, **kwargs):
    print(f"My kwargs from {prompt}: {kwargs}")
    formatted = kwargs.pop("formatted", False)
    print("My kwargs can NOT have formatted value because I am so great it is always false:", formatted)


old_good_function("Old function without formatted param:")
old_good_function("Old function with formatted param:", formatted="stupid overshadowing should be disallowed")

your_stupid_definition("New function without formatted param:")
your_stupid_definition("New function with formatted param:", formatted="stupid overshadowing should be disallowed")

your_stupid_definition("New function without formatted param:", formatted=True, penis="chyba mnie popierdoli")
your_stupid_definition("New function with formatted param:", formatted="stupid overshadowing should be disallowed")

#your_stupid_definition("New function without formatted param:", formatted=True, kutas="chyba mnie pojebie")
#your_stupid_definition("New function with formatted param:", formatted=True, formatted="seriosuly what to do, as that will result with keyword argument repeated: formatted")
```
 same in line https://github.com/run-llama/llama_index/pull/7343/files/a2debb75d030ed33f82bc3286b837782a4d7724f#diff-9b76b57fc39aa4c4e4c6a3ae47ee15dc5f9d38a5e15354aebf20bfcd64902d35R203 

IT WILL BE FALSE always because you cannot PASS it to KWARGS.

you are checking always something that is not exisiting in KWARGS because on interpreter level it will give you `SyntaxError: keyword argument repeated: formatted`

so to sum up, no it is not "working as intended"

**Comment by Disiok** [comment]
Hey @piotrmasior , thanks for raising the concern for the potential bug. We will take a closer look. 


In any case, your behavior is rather rude and not to be tolerated. As such, we are going to block you from our repos.
